src/main/java/uk/gov/hmcts/reform/ccd/ApplicationExecutor.java

```diff
@@ -6,85 +6,59 @@
 import uk.gov.hmcts.reform.ccd.data.model.CaseData;
 import uk.gov.hmcts.reform.ccd.data.model.CaseFamily;
 import uk.gov.hmcts.reform.ccd.parameter.ParameterResolver;
-import uk.gov.hmcts.reform.ccd.service.CaseDeletionResolver;
+import uk.gov.hmcts.reform.ccd.service.CaseDeletionLoggingService;
 import uk.gov.hmcts.reform.ccd.service.CaseDeletionService;
 import uk.gov.hmcts.reform.ccd.service.CaseFinderService;
+import uk.gov.hmcts.reform.ccd.util.ProcessedCasesRecordHolder;
 import uk.gov.hmcts.reform.ccd.util.log.CaseFamiliesFilter;
 
-import java.util.ArrayList;
 import java.util.List;
-import java.util.function.Function;
-import java.util.stream.Collectors;
+import java.util.Set;
 
-import static uk.gov.hmcts.reform.ccd.util.CaseFamilyUtil.FLATTEN_CASE_FAMILIES_FUNCTION;
-import static uk.gov.hmcts.reform.ccd.util.CaseFamilyUtil.POTENTIAL_MULTI_FAMILY_CASE_AGGREGATOR_FUNCTION;
+import static uk.gov.hmcts.reform.ccd.util.CaseFamilyUtil.getCaseData;
 
 @Slf4j
 @Named
 @RequiredArgsConstructor
 public class ApplicationExecutor {
     private final CaseFinderService caseFindingService;
-    private final CaseDeletionResolver caseDeletionResolver;
     private final CaseDeletionService caseDeletionService;
     private final CaseFamiliesFilter caseFamiliesFilter;
     private final ParameterResolver parameterResolver;
+    private final ProcessedCasesRecordHolder processedCasesRecordHolder;
+    private final CaseDeletionLoggingService caseDeletionLoggingService;
 
     public void execute() {
         log.info("Case-Disposer started...");
         final List<CaseFamily> caseFamiliesDueDeletion = caseFindingService.findCasesDueDeletion();
         final List<CaseFamily> deletableCasesOnly = caseFamiliesFilter.getDeletableCasesOnly(caseFamiliesDueDeletion);
-        final List<CaseFamily> deletableLinkedFamiliesSimulation = caseFamiliesFilter
-            .geSimulationCasesOnly(caseFamiliesDueDeletion);
-        final List<CaseFamily> actuallyDeletableCases = new ArrayList<>();
+        final List<CaseFamily> deletableLinkedFamiliesSimulation = caseFamiliesFilter.geSimulationCasesOnly(
+            caseFamiliesDueDeletion);
 
-        final List<CaseData> flattenedCaseFamiliesView = FLATTEN_CASE_FAMILIES_FUNCTION.apply(deletableCasesOnly);
+        Integer requestLimit = parameterResolver.getRequestLimit();
 
-        final List<CaseData> potentialMultiFamilyCases =
-            POTENTIAL_MULTI_FAMILY_CASE_AGGREGATOR_FUNCTION.apply(deletableCasesOnly);
+        final Set<CaseData> allDeletableCases = getCaseData(deletableCasesOnly);
+        final Set<CaseData> simulatedCases = getCaseData(deletableLinkedFamiliesSimulation);
 
-        Integer requestLimit = parameterResolver.getRequestLimit();
+        processedCasesRecordHolder.setSimulatedCases(simulatedCases);
```

Comment:
please move this setter to CaseDeletionLoggingService and populate it with CaseData view. The only purpose this is to make sure that the simulated cases have been processed and logged